test/test_scalar_parameters.py

```diff
@@ -54,8 +54,8 @@ def sinewave2D(comm):
     return (L, hm, top)
 
 
-def test_rms_curvature(comm_self):
-    L, hm, top = sinewave2D(comm_self)
+def test_rms_curvature(comm):
```

Comment:
I think you need to correct this in the functions below as well.

I think `comm_self` should actually disapear from the whole code. 
We should delete the fixture (defined in conftest.py)

Reply:
Okay, I agree. Will delete the fixture. I always thought that this would simply only run on 1 MPI process, but that does not seem to be the case.

Reply:
Should be resolved